### PR TITLE
Add debug option for RL dataset preprocessing

### PR DESCRIPTION
**Pull request by abhinand5**
This PR addresses an issue with the `--debug` flag not functioning as expected for DPO datasets during preprocessing. Despite being specified, the flag failed to produce any debugging output, a problem identified in issue #1400.

# Description

- Enhanced the `load_rl_datasets` function with support for the `--debug` flag and the `cfg.debug` configuration, enabling detailed debug information during the preprocessing of rl/dpo datasets.
- Modified `check_dataset_labels` with an additional parameter `rl: bool`. When set to true, it triggers the use of `check_rl_example_labels` for detailed debugging output, including token ids.
- Added two utility functions to streamline code and reduce redundancy:
    - `color_token_for_rl_debug`: Colors prompt parts based on their type for clearer debug output.
    - `process_tokens_for_rl_debug`: Aids in tokenizing inputs specifically for debugging purposes.

## How has this been tested?

The functionality was verified across three major DPO dataset scenarios to ensure comprehensive coverage:

**Intel Configuration:**
```yaml
rl: dpo
- path: Intel/orca_dpo_pairs
  split: train
  type: chatml.intel
```

**Argilla Configuration:**
```yaml
rl: dpo
datasets:
  - path: argilla/ultrafeedback-binarized-preferences
    split: train
    type: chatml.argilla
```

**Custom Dataset Configuration:**
```yaml
rl: dpo
- path: abhinand/dpo-mix-7k-axolotl
    split: train
    type:
        field_prompt: prompt
        field_chosen: chosen
        field_rejected: rejected
```

## Social Handles (Optional)

- [Twitter](https://twitter.com/abhinand58)

### CODE REVIEW

File: src/axolotl/cli/__init__.py [hunk 1]
```diff
@@ -418,6 +418,23 @@ def load_rl_datasets(
...
+            train_dataset.select(
+                [
+                    random.randrange(0, len(train_dataset) - 1)  # nosec
+                    for _ in range(cli_args.debug_num_examples)
+                ]
+            ),
+            tokenizer,
+            num_examples=cli_args.debug_num_examples,
+            text_only=cli_args.debug_text_only,
+            rl=True,
```
Comment (winglian) [inline src/axolotl/cli/__init__.py:435]:
```suggestion
            rl=bool(cfg.rl),
```

**Review by winglian — APPROVED** [review]
thanks! lgtm

**Comment by NanoCode012** [comment]
After these tests pass, I think this can be merged.

**Comment by NanoCode012** [comment]
Thank you for the PR!

**Pull request merged**